.teamcity/settings.kts

```diff
@@ -37,275 +37,284 @@ version = "2020.1"
+
+				# Run packages tests
+				JEST_JUNIT_OUTPUT_DIR="./test_results/packages" yarn test-packages --maxWorkers=${'$'}JEST_MAX_WORKERS --ci --reporters=default --reporters=jest-junit --silent
+
+				# Run server tests
+				JEST_JUNIT_OUTPUT_DIR="./test_results/server" yarn test-server --maxWorkers=${'$'}JEST_MAX_WORKERS --ci --reporters=default --reporters=jest-junit --silent
+
+				# Run Editing Toolkit tests
+				cd apps/editing-toolkit
+				JEST_JUNIT_OUTPUT_DIR="../../test_results/editing-toolkit" yarn test:js --reporters=default --reporters=jest-junit  --maxWorkers=${'$'}JEST_MAX_WORKERS
+			""".trimIndent()
+			dockerImagePlatform = ScriptBuildStep.ImagePlatform.Linux
+			dockerPull = true
+			dockerImage = "%docker_image%"
+			dockerRunParameters = "-u %env.UID%"
+		}
+		script {
+			name = "Build artifacts"
+			scriptContent = """
+				set -e
+				export HOME="/calypso"
+				export NODE_ENV="test"
+
+				# Update node
+				. "${'$'}NVM_DIR/nvm.sh"
+
+				# Build o2-blocks
+				(cd apps/o2-blocks/ && yarn build --output-path="../../artifacts/o2-blocks")
+
+				# Build wpcom-block-editor
+				(cd apps/wpcom-block-editor/ && yarn build --output-path="../../artifacts/wpcom-block-editor")
+
+				# Build notifications
+				(cd apps/notifications/ && yarn build --output-path="../../artifacts/notifications")
+			""".trimIndent()
+			dockerImagePlatform = ScriptBuildStep.ImagePlatform.Linux
+			dockerPull = true
+			dockerImage = "%docker_image%"
+			dockerRunParameters = "-u %env.UID%"
+		}
+	}
+
+	triggers {
+		vcs {
+			branchFilter = """
+				+:pull/*
+				+:master
+				+:trunk
+			""".trimIndent()
+		}
+	}
+
+	features {
+		feature {
+			type = "xml-report-plugin"
+			param("xmlReportParsing.reportType", "junit")
+			param("xmlReportParsing.reportDirs", "test_results/**/*.xml")
+		}
+		perfmon {
+		}
+		pullRequests {
+			vcsRootExtId = "${WpCalypso.id}"
+			provider = github {
+				serverUrl = ""
+				authType = token {
+					token = "credentialsJSON:57e22787-e451-48ed-9fea-b9bf30775b36"
+				}
+				filterAuthorRole = PullRequests.GitHubRoleFilter.MEMBER_OR_COLLABORATOR
+			}
+		}
+		commitStatusPublisher {
+			vcsRootExtId = "${WpCalypso.id}"
+			publisher = github {
+				githubUrl = "https://api.github.com"
+				authType = personalToken {
+					token = "credentialsJSON:57e22787-e451-48ed-9fea-b9bf30775b36"
+				}
+			}
+		}
+	}
 })
 
 object WpCalypso : GitVcsRoot({
-    name = "wp-calypso"
-    url = "git@github.com:Automattic/wp-calypso.git"
-    pushUrl = "git@github.com:Automattic/wp-calypso.git"
-    branch = "refs/heads/try/teamcity"
-    branchSpec = "+:refs/heads/*"
-    useTagsAsBranches = true
-    authMethod = uploadedKey {
-        uploadedKey = "Sergio TeamCity"
-    }
+	name = "wp-calypso"
+	url = "git@github.com:Automattic/wp-calypso.git"
+	pushUrl = "git@github.com:Automattic/wp-calypso.git"
+	branch = "refs/heads/master"
```

Comment:
Sets the default branch to `master`.

AFAIK this is used for two things:

- When someone changes the config in the UI, TeamCity will make a commit to this branch with the new config, but it will fail because `master` is protected. Most people don't have access to change the config, so this should be fine. Those that have access will see an error message and will have to manually update this file.

- In the UI this will be used to show how much a branch has diverged from `master`.

Reply:
I wonder if we shouldn't have a review process for this, i.e., have TC commit to a branch that we then open PRs from into master, so that we can code review these changes without the hassle of decoding an error message?